tools/gulp/tasks/screenshots.ts

```diff
@@ -5,42 +5,41 @@ import * as admin from 'firebase-admin';
 import * as firebase from 'firebase';
 import {
   openScreenshotsBucket,
-  openFirebaseScreenshotsDatabase,
   connectFirebaseScreenshots} from '../util/firebase';
-import {setGithubStatus} from '../util/github';
+import {isTravisPushBuild} from '../util/travis-ci';
 
 const imageDiff = require('image-diff');
 
+const SCREENSHOT_DIR = './screenshots';
+const LOCAL_GOLDENS = path.join(SCREENSHOT_DIR, `golds`);
+const LOCAL_DIFFS = path.join(SCREENSHOT_DIR, `diff`);
+
 // Directory to which untrusted screenshot results are temporarily written
 //   (without authentication required) before they are verified and copied to
 //   the final storage location.
 const TEMP_FOLDER = 'untrustedInbox';
-const SCREENSHOT_DIR = './screenshots';
 const FIREBASE_REPORT = `${TEMP_FOLDER}/screenshot/reports`;
 const FIREBASE_IMAGE = `${TEMP_FOLDER}/screenshot/images`;
-const FIREBASE_FILELIST = 'screenshot/filenames';
+const FIREBASE_DATA_GOLDENS = `screenshot/goldens`;
+const FIREBASE_STORAGE_GOLDENS = 'goldens';
 
 /** Task which upload screenshots generated from e2e test. */
 task('screenshots', () => {
   let prNumber = process.env['TRAVIS_PULL_REQUEST'];
-  if (prNumber) {
+
+  if (isTravisPushBuild()) {
+    // Only update goldens for build
```

Comment:
Also, it might make sense to rename `isTravisPushBuild` to `isTravisMasterBuild` (or invert the function and rename it to `isTravisPullRequestBuild`)